bb.edn

```diff
@@ -0,0 +1,82 @@
+{
+ :paths ["src"]
+ :deps {zprint/zprint  {:mvn/version "1.2.9"}}
+ :tasks
+ {
+  :requires
+  [[build.build :as build]
+   [build.recurse-keys :as recurse]
+   [babashka.fs :as bfs]
+   [godot-java.core :as godot-java]
+   [babashka.nrepl.server :as srv]
+   ]
+
+  :init
+  (do
+    (def opts {:pre-start-fn build/echo})
+    (def godot-java-jar "target/godot_java-1.0-SNAPSHOT-jar-with-dependencies.jar")
+    (def vars ;; being data allows other tasks to manipulate this map more easily
+      {:JAVA_HOME (build/java-home)
+       :INCLUDES ["-I":JAVA_HOME"/include -I":JAVA_HOME"/include/":OS]
+       :LIBS ["-ljvm -L ":JAVA_HOME"/lib -L ":JAVA_HOME"/lib/server"]
+
+       :compile ["gcc -g -Wall -fPIC ":INCLUDES" -c src/entry.c -o entry.o"]
+       :link ["gcc -g -fPIC ":INCLUDES" ":LIBS" -shared -o entry.so entry.o -rdynamic"]})
+    )
+
+  make-macos
+  {:doc "Equivalent of make for macos"
+   :task (let [{:keys [compile link]} (-> vars (merge {:OS "darwin"})
+                                          (recurse/recurse-keys) #_ (doto build/zpprint))]
+           (shell opts compile)
+           (shell opts link)
+           (bfs/delete "entry.o")
+           (bfs/copy "entry.so" "src/dummy_godot_project/" {:replace-existing true})
+           )}
+
+  generate-and-save-java-classes
+  {:doc "Generate files like godot_java.Godot.GodotBridge"
+   :task
+   (do
+     (println "Generating java classes.")
+     (godot-java/generate-and-save-java-classes))}
+
+  mvn
+  {:depends [generate-and-save-java-classes]
+   :task
+   (shell opts "mvn clean assembly:assembly -DdescriptorId=jar-with-dependencies")}
+
+  run-checks
+  {:task
+   (do
+     (cond
+       (-> godot-java-jar bfs/exists? not)
+       (build/exit (str "The jar has not been generated: " godot-java-jar " .\n Run bb mvn"))
+
+       (-> "src/dummy_godot_project/entry.so" bfs/exists? not)
+       (build/exit (str "The file entry.so does not exist .\n Run bb make-macos"))
+       )
+     (println "All checks passed!"))
+   }
+
+  run-godot
+  {:doc "just run godot"
+   :depends [run-checks]
+   :task (let [;; TODO https://github.com/babashka/fs/pull/133
+               envs {;; TODO Maybe change build to use rpath instead
```

Comment:
So this all started because `LD_LIBRARY_PATH` did not work on MacOS and I didn't know why. Lots of time wasted trying different combinations! 

I then found commentary that using `LD_LIBRARY_PATH` is considered a hack, 
https://prefetch.net/articles/linkers.badldlibrary.html

and it's preferable to tell the executable which paths to look in relatively `-rpath`.

However at this point it gets complicated because relative paths don't work the same on a  Mac as on Linux https://www.youtube.com/watch?v=OL7JTiJs-d8

This is actually what caused me to start trying to build a different `bb` build pipeline for building the `entry.so` for Mac and experiment with `-rpath`.  I got it to build but not to run.

I then discovered `DYLD_LIBRARY_PATH` and it was just simple to use that! I left a note about rpath to revisit given 'hack' considerations.

Thoughts? 
I'm not sure if you want all this as a comment in the code.

By the way, during this investigation, I found out that others have built cross platform docker pipelines for building GDExtensions and this is probably the canonical way to fan them out https://www.youtube.com/watch?v=qhGp8YbzJtU . This would support more platforms still (and allow you to build for non native platforms), but given I managed to fumble my way across the finishing line for MacOS I thought best to leave as a note for future visitation.